lightning/src/util/enforcing_trait_impls.rs

```diff
@@ -72,12 +89,26 @@ impl ChannelKeys for EnforcingChannelKeys {
 		Ok(self.inner.sign_counterparty_commitment(commitment_tx, secp_ctx).unwrap())
 	}
 
-	fn sign_holder_commitment<T: secp256k1::Signing + secp256k1::Verification>(&self, commitment_tx: &HolderCommitmentTransaction, secp_ctx: &Secp256k1<T>) -> Result<Signature, ()> {
-		self.verify_holder_commitment_tx(commitment_tx, secp_ctx);
+	fn sign_holder_commitment<T: secp256k1::Signing + secp256k1::Verification>(&self, holder_commitment_tx: &HolderCommitmentTransaction, secp_ctx: &Secp256k1<T>) -> Result<Signature, ()> {
+		let trusted_tx = self.verify_holder_commitment_tx(holder_commitment_tx, secp_ctx);
 
 		// TODO: enforce the ChannelKeys contract - error if this commitment was already revoked
 		// TODO: need the commitment number
-		Ok(self.inner.sign_holder_commitment(commitment_tx, secp_ctx).unwrap())
+		let revoked = self.revoked_commitment.lock().unwrap();
+		let keys = trusted_tx.keys();
+		if keys.per_commitment_point != self.inner.get_per_commitment_point(*revoked - 1, secp_ctx) {
+			if keys.per_commitment_point != self.inner.get_per_commitment_point(*revoked - 2, secp_ctx) {
+				if keys.per_commitment_point == self.inner.get_per_commitment_point(*revoked, secp_ctx) {
+					println!("attempted to sign the latest revoked local commitment {}", self.inner.commitment_seed[0]);
```

Comment:
I thinik we can just panic with the string here and below - no need to return anything, it should never be hit in testing or fuzzing (or at all).

Reply:
Done, and also rewrote this since we have the actual commitment number given phase 2 signer API.

Reply:
This change was reversed based on further conversation.